classes/Currency.php

```diff
@@ -358,6 +358,27 @@ public function isInstalled($currencyId = null, $shopId = null)
         return (bool) Db::getInstance(_PS_USE_SQL_SLAVE_)->getValue($sql);
     }
 
+    /**
+     * Returns the name of the currency (using the translated name base on the id_lang
+     * provided on creation). This method is useful when $this->name contains an array
+     * but you still need to get its name as a string.
+     *
+     * @return string
+     */
+    public function getName()
+    {
+        if (is_string($this->name)) {
+            return $this->name;
+        }
+
+        $id_lang = $this->id_lang;
+        if ($id_lang === null) {
+            $id_lang = Configuration::get('PS_LANG_DEFAULT');
+        }
+
+        return Tools::ucfirst($this->name[$id_lang]);
```

Comment:
any reason to use ucfirst here and not above?

Reply:
It seems like **$this->name** has already been set in the constructor using Tools:Ucfirst().  I'm not sure if applying the transformation again in a getter context is overkill or not though...  It seems fine to me as it is.

Reply:
> > LGTM
> 
> Hmmm nope, as I said privately in multistore Store A and B can share orders: this is a feature we don't want to break.
> 
> Did someone checked what was the behavior in 1.7.5 in the same conditions?

The behavior in 175 is that you can see an order from another shop if you have the url, although when you list them you don't see it in the list which doesn't seem logical If you switch to a specific shop you should only see this shop's orders
Even though it worked in previous versions it doesn't mean it's the right behavior, this is the kind of bug that became a feature 😅

But the weird thing is that when you export invoices they are shared between the shops... So the two behaviors are contradictory

So we decided not to change the behavior for now, and CLDR is now able to display any installed currency regardless of the shop context But the orders behavior will be discussed later with the multi shop refacto

Reply:
if `$this->name` is a string then it means it already has been formatted in the constructor, so no need for an additional ucfirst
However in `EditCurrencyHandler` the `$currency->name` field is filled with an array, which is logical to allow updating the translated fields, but it caused an error when `DefaultCurrencyInMultiShopException` was thrown because the expected string field was then an array, hence this new getter

Reply:
This method is useful but not a good idea because of the hidden context (aka the current language being inferred magically).

The proper getter would be `getName($idLang)` where `$idLang` cannot be null.

Reply:
Yes except that if `$this->name` doesn't contain the array the method may return a string that doesn't match the asked language, which can be confusing..
Unless we store the localized names in another field `$this->names` in the constructor, similar to what I had to do in another PR:
https://github.com/PrestaShop/PrestaShop/blob/06e1ad1711cfaacd91a94ea6cddf1939b8f762d1/classes/Currency.php#L223

This would allow to get the name in any language regardless of the context

Reply:
Argh, you're right. This sucks.